internal/convert.go

```diff
@@ -339,6 +342,19 @@ func (conv *Conv) StatsAddBadRow(srcTable string, b bool) {
 	}
 }
 
+func PrNodeType(n nodes.Node) string {
```

Comment:
Apologies this is causing churn.

As discussed on PR#60, the original code was kind of broken: we should not have been referencing nodes.Node in convert.go, and it shouldn't have been part of the convert.go APIs or used in convert_test.go.
In fact the only reason that we referenced nodes.Node in convert_test was because we needed to construct a nodes.Node to pass to conv.errorInStatement, conv.schemaStatement etc in TestStatments and TestStatementErrors. And even worse, we needed to call pg_query_go in convert_test.go.

The convert.go and convert_test.go files should be database-agnostic, and shouldn't depend on anything pg related.

I think the way forward is to:
a) Move prNodeType and prNodes back to pgdump.go (sorry for the work!)
b) Change convert_test.go to just use strings (that's effectively what prNodes(parse(t, "CREATE TABLE cart (pid text NOT NULL);")) is doing for us. In fact the test doesn't depend on the content of the string. It just counts the number of entries. Suggest changing TestStatements to:

func TestStatements(t *testing.T) {
	conv := MakeConv()
	conv.ErrorInStatement("ErrorStatement")
	conv.SchemaStatement("SchemaStatement")
	conv.DataStatement("DataStatement")
	conv.SkipStatement("SkipStatement")
	assert.Equal(t, int64(4), conv.Statements())
}

and TestStatementErrors to:

func TestStatementErrors(t *testing.T) {
	conv := MakeConv()
	conv.ErrorInStatement("ErrorStatement")
	assert.Equal(t, int64(1), conv.StatementErrors())
}

Reply:
Agreed. I will go ahead and make the changes